src/index.tsx

```diff
@@ -327,30 +420,62 @@ const styles = StyleSheet.create({
   flexOne: {
     flex: 1,
   },
-  flatList: {
+  flatListH: {
     flex: 1,
     flexDirection: isAndroidRTL ? 'row-reverse' : 'row',
   },
-  paginationContainer: {
+  flatListV: {
+    flex: 1,
+  },
+  paginationContainerH: {
     position: 'absolute',
     bottom: 16,
     left: 16,
     right: 16,
     justifyContent: 'center',
   },
-  paginationDots: {
+  paginationContainerV: {
+    position: 'absolute',
+    bottom: 16,
+    right: 8,
+    top: 16,
+    justifyContent: 'center',
+  },
+  paginationSafeAreaH: {
+
+  },
+  paginationSafeAreaV: {
+    position: 'absolute',
+    right: 8,
+    bottom: 16,
+    top: 16,
+    justifyContent: "center"
+  },
+  paginationDotsH: {
     height: 16,
     margin: 16,
     flexDirection: isAndroidRTL ? 'row-reverse' : 'row',
     justifyContent: 'center',
     alignItems: 'center',
   },
-  dot: {
+  paginationDotsV: {
+    width: 16,
+    margin: 10,
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  dotH: {
     width: 10,
     height: 10,
     borderRadius: 5,
     marginHorizontal: 4,
   },
+  dotV: {
```

Comment:
We could also just have a single `dot` class

Reply:
for vertical render we have to change flex direction of dots

Reply:
But these styles are just the dot itself and doesn't touch any flex direction?

Reply:
Ohk I got it
i'll use single dot class